sql/handler.h

```diff
@@ -3624,7 +3635,8 @@ const char *ha_legacy_type_name(legacy_db_type legacy_type);
 const char *get_canonical_filename(handler *file, const char *path,
                                    char *tmp_path);
 bool mysql_xa_recover(THD *thd);
-
+bool can_hold_read_locks_on_select(THD *thd, thr_lock_type lock_type);
```

Comment:
static (looks like a helper method which has little to do with handler, and not that much even with THD)